### PR TITLE
Refactored ROV Control

### PR DESCRIPTION
**Pull request by Hermanoid**
Good Stuff?

### CODE REVIEW

File: src/rov/rov_control/rov_control/JoySubscriber.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        self.button_7 = False
+        self.button_8 = False
+        self.button_9 = False
+        self.button_10 = False
+        self.button_11 = False
+        self.button_12 = False
+        self.hat_up = False
+        self.hat_down = False
+        self.hat_left = False
+        self.hat_right = False
```
Comment (yameatmeyourdead) [inline src/rov/rov_control/rov_control/JoySubscriber.py:23]: Make this a dict

File: src/rov/rov_control/rov_control/JoySubscriber.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        self.hat_up = msg.axes[mapping["hat_y"]] > 0
+        self.hat_down = msg.axes[mapping["hat_y"]] < 0
+        self.hat_left = msg.axes[mapping["hat_x"]] < 0
+        self.hat_right = msg.axes[mapping["hat_x"]] > 0
+        self.roll = self.apply_deadzone(msg.axes[mapping["roll"]], deadzone)
+        self.pitch = self.apply_deadzone(msg.axes[mapping["pitch"]], deadzone)
+        self.yaw = self.apply_deadzone(msg.axes[mapping["yaw"]], deadzone)
+        self.throttle = self.apply_deadzone(msg.axes[mapping["throttle"]], deadzone)
+        self.hat_x = self.apply_deadzone(msg.axes[mapping["hat_x"]], deadzone)
+        self.hat_y = self.apply_deadzone(msg.axes[mapping["hat_y"]], deadzone)
```
Comment (yameatmeyourdead) [inline src/rov/rov_control/rov_control/JoySubscriber.py:51]: Make this a dict

File: src/rov/rov_control/rov_control/JoySubscriber.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        self.joy_subscription = node.create_subscription(Joy, "joy", self.joystick_callback, 2)
+
+    def now_seconds(self):
+        return self.node.get_clock().now().nanoseconds / 1000000000.0
+
+    def joystick_callback(self, msg: Joy):
+        # self.joystick = ParsedJoy(msg)
+        joystick = ParsedJoy(msg, self.mapping, deadzone=self.deadzone)
+
+        for button in dir(self.toggled_buttons):
```
Comment (yameatmeyourdead) [inline src/rov/rov_control/rov_control/JoySubscriber.py:78]: change this function to loop over the previously created dict

**Review by yameatmeyourdead — CHANGES_REQUESTED** [review]
Refactor garbage variables to dictionaries. Otherwise LGTM

**Review by Hermanoid — COMMENTED** [review]
APPROVE IT

**Review by yameatmeyourdead — APPROVED** [review]

**Pull request merged**